AirCasting/SessionViews/StandaloneSessionCardView.swift

```diff
@@ -46,52 +46,67 @@ struct StandaloneSessionCardView: View {
             Text(Strings.StandaloneSessionCardView.description)
                 .multilineTextAlignment(.center)
             finishAndSyncButton
-                .alert(isPresented: $showingFinishAndSyncAlert) {
-                    finishAndSyncAlert(sessionName: session.name)
-                }
             finishAndDontSyncButton
-                .alert(isPresented: $showingFinishAlert) {
-                    SessionViews.finishSessionAlert(sessionStopper: sessionStopperFactory.getSessionStopper(for: session), sessionName: session.name)
-                }
             .padding()
         }
+        .alert(item: $alert, content: { alert in
+            if alert.id == .finishSessionAlert {
+                return Alert(title: alert.title,
+                             message: alert.message,
+                             primaryButton: .default(alert.buttonTitle, action: {
+                    finishSessionAlertAction(sessionStopper: sessionStopperFactory.getSessionStopper(for: session))
+                }),
+                             secondaryButton: .cancel())
+            } else if alert.id == .finishSessionAndSyncAlert {
+                return Alert(title: alert.title,
+                             message: alert.message,
+                             primaryButton: .default(alert.buttonTitle, action: {
+                    finishSessionAndSyncAlertAction()
+                }), secondaryButton: .cancel())
+            } else {
+                return Alert(title: alert.title,
+                             message: alert.message,
+                             dismissButton: .default(alert.buttonTitle))
+            }
+        })
         .padding()
     }
     
     var finishAndSyncButton: some View {
         Button(Strings.StandaloneSessionCardView.finishAndSyncButtonLabel) {
-            showingFinishAndSyncAlert = true
+            if networkChecker.connectionAvailable {
```

Comment:
This seems to allow to run sync when there is no wifi but there is a cellular internet. Could you make sure if that's what we want to happen? I'm remember vaguely Michael didn't want to run sync on cellular data automatically, without user knowing, but we should double check.

Reply:
Nevermind, I asked Michael and it is okay to allow users to sync over cellular data